aptos-move/framework/src/aptos.rs

```diff
@@ -12,34 +12,80 @@ use move_deps::{
     move_package::compilation::compiled_package::CompiledPackage,
 };
 use once_cell::sync::Lazy;
-use std::collections::BTreeMap;
+use std::collections::{BTreeMap, HashSet};
+use std::hash::Hash;
 
 const APTOS_FRAMEWORK_DIR: &str = "aptos-framework/sources";
 const APTOS_STDLIB_DIR: &str = "aptos-stdlib/sources";
 const MOVE_STDLIB_DIR: &str = "move-stdlib/sources";
+const TOKEN_MODULES_DIR: &str = "aptos-token/sources";
 static APTOS_PKG: Lazy<CompiledPackage> = Lazy::new(|| super::package("aptos-framework"));
+static TOKEN_PKG: Lazy<CompiledPackage> = Lazy::new(|| super::package("aptos-token"));
+static APTOS_STDLIB_PKG: Lazy<CompiledPackage> = Lazy::new(|| super::package("aptos-stdlib"));
+
+pub fn dedup<T, K: Hash + Eq, F: Fn(&T) -> K>(lists: Vec<T>, f: F) -> Vec<T> {
+    let mut res = vec![];
+    let mut keys = HashSet::new();
+    for l in lists {
+        let key: K = f(&l);
+        if keys.insert(key) {
+            res.push(l);
+        }
+    }
+    res
+}
 
 pub fn files() -> Vec<String> {
     let mut files = super::move_files_in_path(MOVE_STDLIB_DIR);
     files.extend(super::move_files_in_path(APTOS_STDLIB_DIR));
     files.extend(super::move_files_in_path(APTOS_FRAMEWORK_DIR));
-    files
+    files.extend(super::move_files_in_path(TOKEN_MODULES_DIR));
+    files.extend(super::move_files_in_path(APTOS_STDLIB_DIR));
+    dedup(files, |f| f.to_string())
 }
 
 pub fn module_blobs() -> Vec<Vec<u8>> {
-    super::module_blobs(&*APTOS_PKG)
+    let mut framework_blobs = super::module_blobs(&*APTOS_PKG);
+    framework_blobs.extend(super::module_blobs(&*TOKEN_PKG));
+    framework_blobs.extend(super::module_blobs(&*APTOS_STDLIB_PKG));
+    dedup(framework_blobs, |blob| {
+        CompiledModule::deserialize(blob).unwrap().self_id()
+    })
```

Comment:
is this necessary? seems ... interesting...

Reply:
the dependencies is the same across packages and duplicated modules loaded. 
without dedup, it will cause issues in publishing